api/src/main/scala/org/virtuslab/ideprobe/log/IdeaLogParser.scala

```diff
@@ -38,10 +39,14 @@ object IdeaLogParser {
   }
 
   private def isNonErrorLogLine(line: String) = {
-    !errorLineRegex.matches(line) && anyLogLineRegex.matches(line)
+    !matches(errorLineRegex, line) && matches(anyLogLineRegex, line)
   }
 
   private def isNotErrorLine(string: String): Boolean = {
-    !errorLineRegex.matches(string)
+    !matches(errorLineRegex, string)
+  }
+
+  private def matches(regex: Regex, string: String): Boolean = {
+    regex.pattern.matcher(string).matches()
```

Comment:
I am wondering what happened here. So it seemed it was enough to just call regex.matches(string) and now we have matches(regex, string) with different implementation. Why so?

Reply:
sadly, `Regex::matches` is not available in scala 2.12